src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSpark.java

```diff
@@ -0,0 +1,522 @@
+                                                           final Path sgaPath,
+                                                           final int threads,
+                                                           final boolean runCorrections)
+            throws IOException{
+
+        final Long breakpointId = fastqOfABreakpoint._1();
+        final String pathStringToRawFASTQ = fastqOfABreakpoint._2().getPath();
+        final String rawFASTQFileName = FilenameUtils.getName(pathStringToRawFASTQ);
+
+        final File workingDir = Files.createTempDirectory( "assembly" + breakpointId.toString() ).toAbsolutePath().toFile();
+        workingDir.deleteOnExit();
+
+        BucketUtils.copyFile(pathStringToRawFASTQ, null, workingDir.toPath().toAbsolutePath().toString()+"/"+rawFASTQFileName);
+
+        final SGAAssemblyResult assembledContigsFileAndRuntimeInfo = runSGAModulesInSerial(sgaPath, workingDir, rawFASTQFileName, threads, runCorrections);
+
+        return new Tuple2<>(breakpointId, assembledContigsFileAndRuntimeInfo);
+    }
+
+    /**
+     * Linear pipeline for running the SGA local assembly process on a particular FASTQ file for its associated putative breakpoint.
+     *
+     * @param sgaPath           full path on the executors to the SGA program
+     * @param workingDir        directory for SGA to work in (since many files are produced along the process)
+     * @param rawFASTQFileName  file name of the FASTQ file, which is assumed to live in workingDir
+     * @param threads           threads to be used by SGA modules, if the module supports multithreading
+     * @param runCorrections    to run SGA correction steps--correct, filter, rmdup, merge--or not
+     * @return                  the result accumulated through running the pipeline, where the contigs could be null if the process erred.
+     */
+    @VisibleForTesting
+    static SGAAssemblyResult runSGAModulesInSerial(final Path sgaPath,
+                                                   final File workingDir,
+                                                   final String rawFASTQFileName,
+                                                   final int threads,
+                                                   final boolean runCorrections){
+
+        int threadsToUse = threads;
+        if( System.getProperty("os.name").toLowerCase().contains("mac") && threads>1){
+            System.err.println("Running on Mac OS X, which doesn't provide unnamed semaphores used by SGA for multithreading. " +
+                               "Resetting threads argument to 1. ");
+            threadsToUse = 1;
+        }
+
+        final File rawFASTQ = new File(workingDir, rawFASTQFileName);
+
+        // the index module is used frequently, so make single instance and pass around
+        final SGAModule indexer = new SGAModule("index");
+        final ArrayList<String> indexerArgs = new ArrayList<>();
```

Comment:
In general better to make variable definitions use the collection interface (`List`) rather than the implementing class (`ArrayList`).